dali/operators/decoder/nvjpeg/nvjpeg_decoder_decoupled_api.h

```diff
@@ -444,24 +459,27 @@ class nvJPEGDecoder : public Operator<MixedBackend>, CachedDecoderImpl {
     }
     const auto &jpeg2k_stream = nvjpeg2k_streams_[data.sample_idx];
     auto ret = nvjpeg2kStreamParse(nvjpeg2k_handle_, input.data(), input.size(),
-                                  0, 0, jpeg2k_stream);
+                                   0, 0, jpeg2k_stream);
     if (ret == NVJPEG2K_STATUS_SUCCESS) {
       nvjpeg2kImageInfo_t image_info;
       NVJPEG2K_CALL(nvjpeg2kStreamGetImageInfo(jpeg2k_stream, &image_info));
       nvjpeg2kImageComponentInfo_t comp;
       NVJPEG2K_CALL(nvjpeg2kStreamGetImageComponentInfo(jpeg2k_stream, &comp, 0));
       uint32_t height = comp.component_height;
       uint32_t width = comp.component_width;
+      data.bpp = std::max<int>(data.bpp, comp.precision);
       for (uint32_t c = 1; c < image_info.num_components; ++c) {
         NVJPEG2K_CALL(nvjpeg2kStreamGetImageComponentInfo(jpeg2k_stream, &comp, c));
         DALI_ENFORCE(height == comp.component_height &&
-                      width == comp.component_width,
-                      make_string("Components dimensions do not match: "
-                                  "component 0 has a shape of {", comp.component_height, ", ",
-                                  comp.component_width, "} and component ", c, " has a shape of {",
-                                  height, ", ", width, "}"));
+                     width  == comp.component_width,
+                     make_string("Components dimensions do not match: "
+                                 "component 0 has a shape of {", comp.component_height, ", ",
+                                 comp.component_width, "} and component ", c, " has a shape of {",
+                                 height, ", ", width, "}"));
+        data.bpp = std::max<int>(data.bpp, comp.precision);
```

Comment:
Hmm.... if precision is defined _per component_, whereas you get max precision. I wonder what would happen in case of an image with different precision for different components... is that even possible?

Reply:
I would put an enforce for that.